tests/accessor/generic_accessor_semantics.cpp

```diff
@@ -79,48 +79,100 @@ DISABLED_FOR_TEST_CASE(hipSYCL)
   int val = 1;
+  int new_val = 2;
+#else
+  constexpr int val = 1;
+  constexpr int new_val = 2;
+#endif
   sycl::buffer<int> buffer{&val, sycl::range<1>{1}};
-  sycl::accessor<int, 1, sycl::access_mode::read_write, sycl::target::host_task>
-      t0{buffer};
 
   SECTION("mutation to copy") {
-    sycl::accessor<int, 1, sycl::access_mode::read_write,
-                   sycl::target::host_task>
-        t1(t0);
-    t1[0] = new_val;
-    CHECK(new_val == t0[0]);
+    {
+      sycl::buffer<int> buffer_result{&result, sycl::range<1>{1}};
+      queue
+          .submit([&](sycl::handler& cgh) {
+            auto acc_result = buffer_result.get_access(cgh);
+            sycl::accessor<int, 1, sycl::access_mode::read_write,
+                           sycl::target::host_task>
+                t0{buffer, cgh};
+            cgh.host_task([=] {
+              t0[0] = val;
+              sycl::accessor<int, 1, sycl::access_mode::read_write,
+                             sycl::target::host_task>
+                  t1(t0);
+              t1[0] = new_val;
+              acc_result[0] = t0[0];
+            });
+          })
+          .wait_and_throw();
+    }
+    CHECK(new_val == result);
   }
 
   SECTION("mutation to original") {
-    sycl::accessor<int, 1, sycl::access_mode::read_write,
-                   sycl::target::host_task>
-        t1(t0);
-    t0[0] = new_val;
-    CHECK(new_val == t1[0]);
+    {
+      sycl::buffer<int> buffer_result{&result, sycl::range<1>{1}};
+      queue
+          .submit([&](sycl::handler& cgh) {
+            auto acc_result = buffer_result.get_access(cgh);
+            sycl::accessor<int, 1, sycl::access_mode::read_write,
+                           sycl::target::host_task>
+                t0{buffer, cgh};
+            cgh.host_task([=] {
+              t0[0] = val;
+              sycl::accessor<int, 1, sycl::access_mode::read_write,
+                             sycl::target::host_task>
+                  t1(t0);
+              t0[0] = new_val;
+              acc_result[0] = t1[0];
+            });
+          })
+          .wait_and_throw();
+    }
+    CHECK(new_val == result);
   }
 
   SECTION("mutation to original, const copy") {
-    const sycl::accessor<int, 1, sycl::access_mode::read_write,
-                         sycl::target::host_task>
-        t1(t0);
-    t0[0] = new_val;
-    CHECK(new_val == t1[0]);
+    {
+      sycl::buffer<int> buffer_result{&result, sycl::range<1>{1}};
+      queue
+          .submit([&](sycl::handler& cgh) {
+            auto acc_result = buffer_result.get_access(cgh);
+            sycl::accessor<int, 1, sycl::access_mode::read_write,
+                           sycl::target::host_task>
+                t0{buffer, cgh};
+            cgh.host_task([=] {
+              t0[0] = val;
+              const sycl::accessor<int, 1, sycl::access_mode::read_write,
+                                   sycl::target::host_task>
+                  t1(t0);
```

Comment:
```suggestion
                  t1{t0};
```
and so on everywhere?

Reply:
Fixed everywhere.